src/models/Call.ts

```diff
@@ -792,70 +789,41 @@ export class ElementCall extends Call {
         super.destroy();
     }
 
+    private onRTCSessionEnded = (roomId: string, session: MatrixRTCSession): void => {
+        if (roomId == this.roomId) {
+            this.destroy();
+        }
+    };
+
     /**
      * Sets the call's layout.
      * @param layout The layout to switch to.
      */
     public async setLayout(layout: Layout): Promise<void> {
         const action = layout === Layout.Tile ? ElementWidgetActions.TileLayout : ElementWidgetActions.SpotlightLayout;
-
         await this.messaging!.transport.send(action, {});
     }
 
+    private onMembershipChanged = (): void => this.updateParticipants();
+
     private updateParticipants(): void {
         const participants = new Map<RoomMember, Set<string>>();
 
-        for (const [member, deviceMap] of this.groupCall.participants) {
-            participants.set(member, new Set(deviceMap.keys()));
+        for (const m of this.session.memberships) {
+            if (m.sender == undefined) continue;
```

Comment:
Is == undefined what you mean here? Either triple equals or just testing for truthiness would be more standard.